### PR TITLE
Add entry only noise injection

### PR DESCRIPTION
**Pull request by daviddpruitt**
# Modulus Pull Request

## Description
Adds entry only noise injection
Adds CRPS noise options to DoubleConvNextBlock

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py [hunk 1]
```diff
@@ -332,6 +348,10 @@ def __init__(
         activation: th.nn.Module = None,
         enable_nhwc: bool = False,
         enable_healpixpad: bool = False,
+        conditional_layer_norm: Callable = None,
+        conditional_layer_norm_once: bool = False,
+        batch_norm: bool = False,
```
Comment (zacespinosa) [inline physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py]: Since we've moved towards CLN and LN as a group, I think if no one is using this functionality we should leave batch_norm out

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py [hunk 2]
```diff
@@ -400,8 +433,20 @@ def __init__(
                 enable_healpixpad=enable_healpixpad,
             )
         )
+
+        # Apply batch norm and conditional layer norm if needed
+        if not conditional_layer_norm_once:
```
Comment (zacespinosa) [inline physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py]: removing batch norm also let's us simplify this logic: `if not conditional_layer_norm_once and conditional_layer_norm is not None:`

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py [hunk 3]
```diff
@@ -386,6 +408,17 @@ def __init__(
                 enable_healpixpad=enable_healpixpad,
             )
 
+        if conditional_layer_norm_once:
```
Comment (zacespinosa) [inline physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py:415]: Can you add a comment for readability here and in each block that uses this logic. Something like this: 

`if conditional_layer_norm_once is true, but conditional_layer_norm is not defined, apply layer norm, otherwise skip. `

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py [hunk 4]
```diff
@@ -332,6 +348,10 @@ def __init__(
         activation: th.nn.Module = None,
         enable_nhwc: bool = False,
         enable_healpixpad: bool = False,
+        conditional_layer_norm: Callable = None,
+        conditional_layer_norm_once: bool = False,
+        batch_norm: bool = False,
+        dropout: float = 0.0,
```
Comment (zacespinosa) [inline physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py:353]: I'm fine with adding `dropout` but is anyone using this? and I'd probably make this a separate PR since it's a different feature.

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py [hunk 5]
```diff
@@ -490,11 +577,44 @@ def forward(self, x):
         -------
         torch.Tensor
             result of the forward pass
+        conditions_cln: torch.Tensor, optional
+            conditions for the conditional layer normalization
         """
+        # save residual for the first block
+        x1_residual = self.skip_module1(x)
```
Comment (zacespinosa) [inline physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py:582]: Just for some context, I changed to apply CLN only to the main tensor and skipping the residual to enable AdaLN and keep things consistent between the two cases. Technically, I never compared doing cln(x_residual + x) vs. cln(x). Could be worth testing this in the future. 

Maybe add a comment here explaining that. Something like: 

TODO: compare cln(x) vs. cln(x1_residual + x)

**Review by zacespinosa — APPROVED** [review]
Looks good. Left a few comments and requests.

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py (hunk 4 above)
Comment (daviddpruitt) [reply physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py:353]: I was running some testing with it. We can drop in the future.

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py (hunk 3 above)
Comment (daviddpruitt) [reply physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py:415]: Added comments, let me know if they make sense.

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py (hunk 1 above)
Comment (daviddpruitt) [reply physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py]: That's a good idea, I pulled out the batch norm. This does bring up that the cln_once flag really does two separate things. We can adjust it later as we figure out which works best.

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py (hunk 2 above)
Comment (daviddpruitt) [reply physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py]: I pulled the norm as you suggested and simplified these.

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py (hunk 5 above)
Comment (daviddpruitt) [reply physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py:582]: I added a comment.

File: physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py (hunk 1 above)
Comment (zacespinosa) [reply physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py]: Sounds good to me. I suggest we had a "TODO: simplify XXX" in the code as a comment so that we don't forget. I think it's generally nice to sprinkle these TODOs through the code. Improves readability and decreases the need to remember everything

Comment (daviddpruitt) [reply physicsnemo/models/dlwp_healpix_layers/healpix_blocks.py]: Added in with some explainer.
